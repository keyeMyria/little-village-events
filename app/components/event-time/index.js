export { default } from "./event-time";
