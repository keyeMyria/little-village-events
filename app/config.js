export default {
    eventThresholds: {
        upcoming: 60,
        past: 60
    }
};
