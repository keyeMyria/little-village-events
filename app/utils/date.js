import _sortBy from "lodash/sortBy";
import startOfDay from "date-fns/start_of_day";
import endOfDate from "date-fns/end_of_day";
import startOfWeek from "date-fns/start_of_week";
import endOfWeek from "date-fns/end_of_week";
import differenceInCalendarDays from "date-fns/difference_in_calendar_days";
import differenceInCalendarWeeks from "date-fns/difference_in_calendar_weeks";
import addDays from "date-fns/add_days";
import addMinutes from "date-fns/add_minutes";
import subDays from "date-fns/sub_days";
import subMinutes from "date-fns/sub_minutes";
import dateFormat from "date-fns/format";

export { default as isAfter } from "date-fns/is_after";
export { default as isBefore } from "date-fns/is_before";
export { default as isSameDay } from "date-fns/is_same_day";

export { default as minutesDiff } from "date-fns/difference_in_minutes";
export { default as hoursDiff } from "date-fns/difference_in_hours";
export const daysDiff = differenceInCalendarDays;

export const format = dateFormat;

export const now = Date.now;


export const dayStart = ( date ) => {
    return startOfDay( date );
};


export const dayEnd = ( date ) => {
    return endOfDate( date );
};


export const dayTimestamp = ( date ) => {
    return dayStart( date ).valueOf();
};


export const weekStart = ( date ) => {
    return startOfWeek( date );
};


export const weekEnd = ( date ) => {
    return endOfWeek( date );
};


export const weekTimestamp = ( date ) => {
    return weekStart( date ).valueOf();
};


export const addToDate = ( date, params ) => {
    const days = params.day || params.days;
    if ( days )
        return addDays( date, days );

    const minutes = params.minute || params.minutes;
    if ( minutes )
        return addMinutes( date, minutes );

    return date;
};


export const subtractFromDate = ( date, params ) => {
    const days = params.day || params.days;
    if ( days )
        return subDays( date, days );

    const minutes = params.minute || params.minutes;
    if ( minutes )
        return subMinutes( date, minutes );

    return date;
};


export const moveTimeToDate = ( time, date ) => {
    if ( time === date )
        return time;
    const daysToAdd = daysDiff( date, time );
    return daysToAdd
        ? addToDate( time, { days: daysToAdd } )
        : time
    ;
};


export const sortByDate = ( items, property ) => {
    return _sortBy( items, item => item[ property ].valueOf() );
};

export const calendarFormat = ( date, referenceDate, formats ) => {
    referenceDate = referenceDate || now();

    const days = daysDiff( date, referenceDate );
    switch ( days ) {
        case -1:
            return format( date, formats.lastDay );
        case 0:
            return format( date, formats.sameDay );
        case 1:
            return format( date, formats.nextDay );
    }

    const weeks = differenceInCalendarWeeks( referenceDate );
    switch ( weeks ) {
        case -1:
            return format( date, formats.lastWeek );
        case 0:
            return format( date, days < 0 ? formats.lastWeek : formats.nextWeek );
        case 1:
            return format( date, formats.nextWeek );
    }

    return format( date, formats.sameElse );
};
