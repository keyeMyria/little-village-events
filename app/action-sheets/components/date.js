import { format } from "app/utils/date";

import React from "react";
import { View, Text } from "react-native";

export default ( { styles, highlighted, date } ) => (
    <View style={ [ styles.date, highlighted && styles.dateHighlighted ] }>
        <Text style={ [ styles.dateLabel, highlighted && styles.dateLabelHighlighted ] }>
            { format( date, "ddd, MMM D" ).toUpperCase() }
        </Text>
    </View>
);
