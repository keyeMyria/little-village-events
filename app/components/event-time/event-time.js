import * as Styles from "../../styles";

import { formatStartTime } from "../../utils/event";
import { eventTense } from "../../utils/event-time";

import React from "react";
import { View, Text } from "react-native";

import _isString from "lodash/isString";


const stylesSize = {
    regular: Styles.create( {
        root: {
            paddingHorizontal: 5,
            paddingVertical: 4,

            borderRadius: 3
        },

        label: {
            fontSize: Styles.variables.regularFontSize
        }

    } ),
    small: Styles.create( {
        root: {
            paddingHorizontal: 4,
            paddingVertical: 3,

            borderRadius: 5
        },

        label: {
            fontSize: Styles.variables.smallFontSize
        }

    } )

};

const colorsByTense = {
    past: Styles.create( {
        root: {
            backgroundColor: Styles.variables.mediumGrayColor
        },
        label: {
            color: Styles.variables.textColor
        }
    } ),
    present: Styles.create( {
        root: {
            backgroundColor: Styles.variables.warningColor
        },
        label: {
            color: Styles.variables.invertedTextColor
        }
    } ),
    upcoming: Styles.create( {
        root: {
            backgroundColor: Styles.variables.highlightColor
        },
        label: {
            color: Styles.variables.invertedTextColor
        }
    } ),
    future: Styles.create( {
        root: {
            backgroundColor: Styles.variables.mediumGrayColor
        },
        label: {
            color: Styles.variables.invertedTextColor
        }
    } ),

};


const EventTime = ( { event, calendarDay, ongoing, size = "small" } ) => {

    const styles = stylesSize[size];
    const tense = ongoing ? "future" : eventTense( event, calendarDay );
    const colors = colorsByTense[ tense ];

    return (
        <View style={ [ styles.root, colors.root ] }>
            <Text style={ [ styles.label, colors.label ] }>
                { formatStartTime( event ) }
            </Text>
        </View>
    );
};

export default EventTime;
