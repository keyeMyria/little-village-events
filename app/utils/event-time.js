import { addToDate, dayStart, dayEnd } from "./date";
import config from "../config";

import moment from "moment";

const { eventThresholds } = config;

const validateRSVPTimeRange = ( event, result, calendarDay ) => {
    const eventDayStart = dayStart( result.startTime );
    const eventDayEnd = dayEnd( result.endTime );

    if ( calendarDay.isBefore( eventDayStart ) || calendarDay.isAfter( eventDayEnd ) )
        throw new Error( `RSVP Time: given calendar day is outside of event days:\n${ JSON.stringify( {
            event,
            result,
            calendarDay
        } ) }` );
};

// TODO remove end time calculation, tense checks should use config.eventThresholds.past instead

const calcRSVPTimeImpl = ( { startTime, endTime, allDay }, calendarDay ) => {
    const calendarDayStart = dayStart( calendarDay );
    const eventFirstDayStart = dayStart( startTime );

    if ( !endTime ) {
        if ( allDay ) {
            endTime = addToDate( eventFirstDayStart, { hours: 17 } );
            if ( endTime.isAfter( startTime ) )
                return { startTime, endTime };
        }

        endTime = addToDate( startTime, { minutes: 15 } );
        const eventFirstDayEnd = dayEnd( startTime );
        if ( endTime.isAfter( eventFirstDayEnd ) )
            endTime = eventFirstDayEnd;
        return { startTime, endTime };
    }

    const lastEventDayStart = dayStart( endTime );

    if ( eventFirstDayStart.isAfter( calendarDayStart ) ) {
        return {
            startTime,
            endTime: addToDate( endTime, { days: eventFirstDayStart.diff( lastEventDayStart, "days" ) } )
        };
    }

    if ( lastEventDayStart.isBefore( calendarDayStart ) ) {
        return {
            startTime: addToDate( startTime, { days: lastEventDayStart.diff( eventFirstDayStart, "days" ) } ),
            endTime
        };
    }

    return {
        startTime: addToDate( startTime, { days: calendarDayStart.diff( eventFirstDayStart, "days" ) } ),
        endTime: addToDate( endTime, { days: calendarDayStart.diff( lastEventDayStart, "days" ) } )
    };
};


export const calcRSVPTime = ( event, calendarDay ) => {
    const result = calcRSVPTimeImpl( event, calendarDay );
    validateRSVPTimeRange( event, result, calendarDay );
    return result;
};


export const rsvpTense = ( rsvp, currentTime ) => {
    currentTime = currentTime || moment();
    const { startTime, endTime } = rsvp;

    if ( endTime.isBefore( currentTime ) )
        return "past";

    if ( currentTime.isBefore( startTime ) ) {
        const minutes = startTime.diff( currentTime, "minutes" );
        return minutes <= eventThresholds.upcoming ? "upcoming" : "future";
    }

    if ( rsvp.allDay )
        return "upcoming";

    return "present";
};


export const isRSVPFeatured = ( rsvp, currentTime ) => {
    return [ "upcoming", "present" ].indexOf( rsvpTense( rsvp, currentTime ) ) > -1;
};


export const eventTense = ( event, calendarDay, currentTime ) => {
    currentTime = currentTime || moment();
    return rsvpTense( calcRSVPTime( event, calendarDay ), currentTime );
};
