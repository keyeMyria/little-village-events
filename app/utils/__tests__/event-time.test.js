import { dayStart, addToDate, subtractFromDate } from "../date";
import { firstRSVPDate, lastRSVPDate, getRSVPInfo, RSVPTimeForDay, calcRSVPTime, eventTense } from "../event-time";

import _keys from "lodash/keys";
import _pick from "lodash/pick";

import sinon from "sinon";
import config from "../../config";

const dateHour = ( date, hour ) => addToDate( dayStart( date ), { minutes: hour * 60 } );

const earlyMorning = date => dateHour( date, 2 );
const morning = date => dateHour( date, 9 );
const noon = date => dateHour( date, 12 );
const evening = date => dateHour( date, 18 );
const lateEvening = date => dateHour( date, 22 );


const integrityWrapper = testFunc =>
    ( event, ...args ) => {
        const eventSnapshot = JSON.stringify( event );
        testFunc( event, ...args );
        expect( JSON.stringify( event ) ).toMatch( eventSnapshot );
    };


const allDayWrapper = testFunc =>
    ( event, ...args ) => {
        testFunc( event, ...args );
        if ( event.allDay === undefined )
            testFunc( { ...event, allDay: true }, ...args );
    };


const expectedWrapper = testFunc =>
    expected => event => testFunc( event, expected );

const allDayExpectedWrapper = ( testFunc ) =>
    expectedWrapper( allDayWrapper( testFunc ) )
;


const toJSON = event => {
    return JSON.stringify( _pick( event, _keys( event ).sort() ) );
};

describe( "event-time", () => {

    const today = noon( new Date( "2018-05-14" ) );
    const tomorrow = addToDate( today, { days: 1 } );
    const weekLater = addToDate( today, { days: 7 } );
    const yesterday = subtractFromDate( today, { days: 1 } );
    const weekAgo = subtractFromDate( today, { days: 7 } );

    const sandbox = sinon.createSandbox();

    beforeEach( () => {
    } );

    afterEach( () => {
        sandbox.restore();
    } );

    describe( "firstRSVPDate", () => {
        const test = integrityWrapper(
            ( event, expected ) => expect( JSON.stringify( firstRSVPDate( event ) ) ).toMatch( JSON.stringify( expected ) )
        );

        it( "handles one time events", () => {
            test( { startTime: morning( today ) }, morning( today ) );
            test( { startTime: morning( today ), allDay: true }, dayStart( today ) );
        } );

        it( "handles reoccurring events", () => {
            test( { startTime: morning( weekAgo ), endTime: evening( weekLater ) }, morning( weekAgo ) );
            test( {
                startTime: morning( weekAgo ),
                endTime: evening( weekLater ),
                allDay: true
            }, dayStart( weekAgo ) );
        } );

    } );

    describe( "lastRSVPDate", () => {
        const test = integrityWrapper(
            ( event, expected ) => expect( JSON.stringify( lastRSVPDate( event ) ) ).toMatch( JSON.stringify( expected ) )
        );

        it( "handles one time events", () => {
            [
                { startTime: morning( today ) },
                { startTime: morning( today ), endTime: evening( today ) },
                { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: false }
            ].forEach( allDayExpectedWrapper( test )( null ) );
        } );

        it( "handles reoccurring events", () => {
            test( { startTime: morning( weekAgo ), endTime: evening( weekLater ) }, morning( weekLater ) );
            test( {
                startTime: morning( weekAgo ),
                endTime: evening( weekLater ),
                allDay: true
            }, dayStart( weekLater ) );
            test(
                { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: true },
                dayStart( tomorrow )
            );

        } );

        it( "handles events with incorrect endTime", () => {
            test( { startTime: evening( today ), endTime: morning( today ) }, null );
        } );

    } );

    describe( "getRSVPInfo", () => {
        const test = integrityWrapper(
            ( event, expected ) => expect( JSON.stringify( getRSVPInfo( event ) ) ).toMatch( JSON.stringify( expected ) )
        );

        it( "handles one time events", () => {
            test(
                { startTime: morning( today ) },
                { first: morning( today ), last: null, allDay: false, duration: 0 }
            );
            test(
                { startTime: morning( today ), endTime: evening( today ) },
                { first: morning( today ), last: null, allDay: false, duration: 540 }
            );

            test(
                { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ) },
                { first: lateEvening( today ), last: null, allDay: false, duration: 240 }
            );

            test(
                { startTime: evening( today ), endTime: morning( today ) },
                { first: evening( today ), last: null, allDay: false, duration: 0 }
            );

        } );

        it( "handles reoccurring events", () => {
            test(
                { startTime: morning( yesterday ), endTime: evening( tomorrow ) },
                { first: morning( yesterday ), last: morning( tomorrow ), allDay: false, duration: 540 }
            );

            test(
                { startTime: morning( yesterday ), endTime: evening( tomorrow ), allDay: true },
                { first: dayStart( yesterday ), last: dayStart( tomorrow ), allDay: true, duration: 0 }
            );


            test(
                { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: true },
                { first: dayStart( today ), last: dayStart( tomorrow ), allDay: true, duration: 0 }
            );

        } );

    } );


    describe( "RSVPTimeForDay", () => {

        const test = integrityWrapper(
            ( event, expected ) => expect( toJSON( RSVPTimeForDay( event, today ) ) ).toMatch( toJSON( expected ) )
        );

        const testAllDayExpected = allDayExpectedWrapper( test );

        describe( "one time events", () => {

            it( "handles events", () => {

                test(
                    { startTime: morning( today ) },
                    { startTime: morning( today ), endTime: null, allDay: false }
                );


                test(
                    { startTime: morning( today ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );


                test(
                    { startTime: morning( today ), endTime: evening( today ) },
                    { startTime: morning( today ), endTime: evening( today ), allDay: false }
                );


                test(
                    { startTime: morning( today ), endTime: evening( today ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );


                [
                    yesterday,
                    tomorrow,
                ].forEach( day => {
                    testAllDayExpected( null )( { startTime: morning( day ) } );
                    testAllDayExpected( null )( { startTime: morning( day ), endTime: evening( day ) } );
                } );

                test( { startTime: lateEvening( yesterday ), endTime: earlyMorning( today ) }, null );


                test(
                    { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ) },
                    { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: false }
                );


            } );


        } );

        describe( "reoccurring events", () => {

            it( "handles past events", () => {
                [
                    { startTime: morning( weekAgo ), endTime: evening( yesterday ) },
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( yesterday ) },
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( today ), allDay: false }
                ].forEach( testAllDayExpected( null ) );

            } );


            it( "handles ongoingEvents events", () => {

                [
                    { startTime: morning( weekAgo ), endTime: evening( weekLater ) },
                    { startTime: morning( weekAgo ), endTime: evening( today ) },
                    { startTime: morning( today ), endTime: evening( weekLater ) },
                ].forEach( event => {
                    test(
                        { ...event, allDay: false },
                        { startTime: morning( today ), endTime: evening( today ), allDay: false }
                    );
                    if ( event.allDay === undefined ) {
                        test(
                            { ...event, allDay: true },
                            { startTime: dayStart( today ), endTime: null, allDay: true } );
                    }
                } );


                test(
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( today ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );


                test(
                    { startTime: lateEvening( today ), endTime: earlyMorning( weekLater ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );


                test(
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( weekLater ) },
                    { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: false }
                );

            } );

            it( "handles future events", () => {

                [
                    { startTime: morning( tomorrow ), endTime: evening( weekLater ) },
                    { startTime: lateEvening( tomorrow ), endTime: earlyMorning( weekLater ) }
                ].forEach( testAllDayExpected( null ) );

            } );

        } );

    } );


    describe( "calcRSVPTime", () => {

        const test = integrityWrapper(
            ( event, expected ) => expect( JSON.stringify( calcRSVPTime( event, today ) ) ).toMatch( JSON.stringify( expected ) )
        );

        describe( "one time events", () => {

            it( "returns rsvp time for current events", () => {
                test(
                    { startTime: morning( today ) },
                    { startTime: morning( today ), endTime: null, allDay: false }
                );

                test(
                    { startTime: morning( today ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );

                test(
                    { startTime: morning( today ), endTime: evening( today ) },
                    { startTime: morning( today ), endTime: evening( today ), allDay: false }
                );

                test(
                    { startTime: morning( today ), endTime: evening( today ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );

                test(
                    { startTime: morning( weekAgo ), endTime: evening( weekLater ) },
                    { startTime: morning( today ), endTime: evening( today ), allDay: false }
                );

                test(
                    { startTime: morning( weekAgo ), endTime: evening( weekLater ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );

                test(
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( weekLater ) },
                    { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: false }
                );

            } );

            it( "throws exceptions for non-current events", () => {

                [
                    yesterday,
                    tomorrow,
                ].forEach( day => {
                    expect( () => calcRSVPTime( { startTime: morning( day ) }, today ) ).toThrow();
                    expect( () => calcRSVPTime( { startTime: morning( day ), allDay: true }, today ) ).toThrow();
                    expect( () => calcRSVPTime(
                        { startTime: morning( day ), endTime: evening( day ) }
                        , today )
                    ).toThrow();
                    expect( () => calcRSVPTime(
                        { startTime: morning( day ), endTime: evening( day ), allDay: true }
                        , today )
                    ).toThrow();
                } );


                expect( () => calcRSVPTime(
                    { startTime: lateEvening( yesterday ), endTime: earlyMorning( today ) }
                    , today )
                ).toThrow();
            } );


        } );


        describe( "reoccurring events", () => {

            it( "returns rsvp time for current events", () => {
                [
                    { startTime: morning( weekAgo ), endTime: evening( weekLater ) },
                    { startTime: morning( weekAgo ), endTime: evening( today ) },
                    { startTime: morning( today ), endTime: evening( weekLater ) }
                ].forEach( event => {
                    test(
                        event,
                        { startTime: morning( today ), endTime: evening( today ), allDay: false }
                    );

                    test(
                        { ...event, allDay: true },
                        { startTime: dayStart( today ), endTime: null, allDay: true }
                    );
                } );


                test(
                    { startTime: earlyMorning( weekAgo ), endTime: morning( weekLater ) },
                    { startTime: earlyMorning( today ), endTime: morning( today ), allDay: false }
                );

                test(
                    { startTime: evening( weekAgo ), endTime: lateEvening( weekLater ) },
                    { startTime: evening( today ), endTime: lateEvening( today ), allDay: false }
                );

                test(
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( weekLater ) },
                    { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: false }
                );

                test(
                    { startTime: lateEvening( yesterday ), endTime: earlyMorning( today ), allDay: true },
                    { startTime: dayStart( today ), endTime: null, allDay: true }
                );

            } );

            it( "throws exceptions for invalid events", () => {
                [
                    { startTime: morning( weekAgo ), endTime: evening( yesterday ) },
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( yesterday ) },
                    { startTime: morning( tomorrow ), endTime: evening( weekLater ) },
                    { startTime: lateEvening( tomorrow ), endTime: earlyMorning( weekLater ) },
                ].forEach( event => {
                    expect( () => calcRSVPTime( event, today ) ).toThrow();
                    expect( () => calcRSVPTime( { ...event, allDay: true }, today ) ).toThrow();
                } );

                expect( () => calcRSVPTime(
                    { startTime: lateEvening( weekAgo ), endTime: earlyMorning( today ) }
                    , today )
                ).toThrow();

            } );


        } );

    } );

    describe( "eventTense", () => {

        const currentTime = noon( today );

        const test = integrityWrapper(
            ( event, expected ) => expect( eventTense( event, today, currentTime ) ).toEqual( expected )
        );
        const testExpected = expectedWrapper( test );
        const testAllDayExpected = allDayExpectedWrapper( test );

        it( "handles one time events", () => {

            [
                { startTime: morning( yesterday ) },
                { startTime: morning( yesterday ), endTime: evening( yesterday ) },
                { startTime: lateEvening( yesterday ), endTime: earlyMorning( today ), allDay: false },
                { startTime: lateEvening( yesterday ), endTime: evening( today ), allDay: false },
                { startTime: earlyMorning( today ), endTime: morning( today ), allDay: false },
                { startTime: morning( today ), allDay: false },
                { startTime: subtractFromDate( currentTime, { minutes: config.eventThresholds.past } ), allDay: false },
            ].forEach( testAllDayExpected( "past" ) );

            [
                { startTime: morning( today ), endTime: evening( today ) },
                { startTime: currentTime, endTime: evening( today ) },
                { startTime: subtractFromDate( currentTime, { minutes: config.eventThresholds.past - 1 } ) },
            ].forEach( testExpected( "present" ) );

            [
                { startTime: morning( today ), allDay: true },
                { startTime: morning( today ), endTime: evening( today ), allDay: true },
                { startTime: evening( today ), allDay: true },
                { startTime: evening( today ), endTime: lateEvening( today ), allDay: true },
                { startTime: addToDate( currentTime, { minutes: config.eventThresholds.upcoming - 1 } ) },
                {
                    startTime: addToDate( currentTime, { minutes: config.eventThresholds.upcoming - 1 } ),
                    endTime: evening( today )
                },
            ].forEach( testExpected( "upcoming" ) );

            [
                { startTime: evening( today ), allDay: false },
                { startTime: evening( today ), endTime: lateEvening( today ), allDay: false },
                { startTime: lateEvening( today ), endTime: earlyMorning( tomorrow ), allDay: false },
                { startTime: morning( tomorrow ) },
                { startTime: morning( tomorrow ), endTime: evening( tomorrow ) },
                { startTime: addToDate( currentTime, { minutes: config.eventThresholds.upcoming } ), allDay: false },
            ].forEach( testAllDayExpected( "future" ) );
        } );

        it( "handles reoccurring events", () => {
            [
                { startTime: morning( weekAgo ), endTime: evening( yesterday ) },
                { startTime: morning( weekAgo ), endTime: morning( today ), allDay: false },
                { startTime: lateEvening( weekAgo ), endTime: earlyMorning( today ), allDay: false },
                { startTime: lateEvening( weekAgo ), endTime: evening( today ), allDay: false },
            ].forEach( testAllDayExpected( "past" ) );

            [
                { startTime: morning( weekAgo ), endTime: evening( weekLater ) },
                { startTime: morning( weekAgo ), endTime: evening( today ) },
                { startTime: morning( weekAgo ), endTime: earlyMorning( weekLater ) },
                { startTime: earlyMorning( today ), endTime: evening( weekLater ) },
            ].forEach( testExpected( "present" ) );

            [
                { startTime: morning( weekAgo ), endTime: evening( weekLater ), allDay: true },
                { startTime: morning( weekAgo ), endTime: evening( today ), allDay: true },
                { startTime: morning( weekAgo ), endTime: earlyMorning( weekLater ), allDay: true },
                { startTime: earlyMorning( today ), endTime: evening( weekLater ), allDay: true },
                { startTime: lateEvening( today ), endTime: morning( weekLater ), allDay: true },
                {
                    startTime: addToDate( currentTime, { minutes: config.eventThresholds.upcoming - 1 } ),
                    endTime: evening( weekLater ),
                    allDay: false
                }
            ].forEach( testExpected( "upcoming" ) );

            [
                { startTime: lateEvening( weekAgo ), endTime: earlyMorning( weekLater ), allDay: false },
                { startTime: lateEvening( weekAgo ), endTime: evening( weekLater ), allDay: false },
                { startTime: evening( weekAgo ), endTime: lateEvening( weekLater ), allDay: false },
                { startTime: evening( today ), endTime: lateEvening( weekLater ), allDay: false },
                { startTime: morning( tomorrow ), endTime: evening( weekLater ) },
            ].forEach( testAllDayExpected( "future" ) );

        } );

    } );

} );
