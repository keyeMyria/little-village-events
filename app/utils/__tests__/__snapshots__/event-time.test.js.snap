// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event-time calcRSVPTime all day events 1`] = `
Object {
  "endTime": "2018-05-14T17:00:00-05:00",
  "startTime": "2018-05-14T09:00:00-05:00",
}
`;

exports[`event-time calcRSVPTime all day events 2`] = `
Object {
  "endTime": "2018-05-14T17:15:00-05:00",
  "startTime": "2018-05-14T17:00:00-05:00",
}
`;

exports[`event-time calcRSVPTime events with end time 1`] = `
Object {
  "endTime": "2018-05-14T17:00:00-05:00",
  "startTime": "2018-05-14T09:00:00-05:00",
}
`;

exports[`event-time calcRSVPTime events with end time 2`] = `
Object {
  "endTime": "2018-05-14T17:00:00-05:00",
  "startTime": "2018-05-14T09:00:00-05:00",
}
`;

exports[`event-time calcRSVPTime events with end time 3`] = `
Object {
  "endTime": "2018-05-14T17:00:00-05:00",
  "startTime": "2018-05-14T09:00:00-05:00",
}
`;

exports[`event-time calcRSVPTime events without end time 1`] = `
Object {
  "endTime": "2018-05-14T09:15:00-05:00",
  "startTime": "2018-05-14T09:00:00-05:00",
}
`;
